.github/workflows/build_and_test.yml

```
@@ -0,0 +1,308 @@
+        type: choice
+        options:
+          - patch
+          - minor
+          - major
+      branch:
+        description: "Branch to build (for adhoc)"
+        required: false
+        default: "main"
+
+# Anchors for common steps
+x-checkout-cache: &checkout-cache
```

actionlint (1.7.4)

unexpected key "x-checkout-cache" for "workflow" section. expected one of "concurrency", "defaults", "env", "jobs", "name", "on", "permissions", "run-name"

(syntax-check)